libminifi/src/RemoteProcessorGroupPort.cpp

```diff
@@ -150,6 +194,87 @@ void RemoteProcessorGroupPort::onTrigger(core::ProcessContext *context, core::Pr
   return;
 }
 
+void RemoteProcessorGroupPort::refreshRemoteSite2SiteInfo() {
+  if (this->host_.empty() || this->port_ == -1 || this->protocol_.empty())
+      return;
+
+  std::string fullUrl = this->protocol_ + this->host_ + ":" + std::to_string(this->port_) + "/nifi-api/controller/";
```

Comment:
the /nifi-api/controller return 
it return the site2site port and we can use that to do the site2site negotiation with master to find the peer info
{
  "revision": {
    "clientId": "d40fb824-070b-4547-9b1c-f50f5ba0a677"
  },
  "controller": {
    "id": "fe4a3a42-53b6-4af1-a80d-6fdfe60de97f",
    "name": "NiFi Flow",
    "comments": "",
    "runningCount": 3,
    "stoppedCount": 2,
    "invalidCount": 0,
    "disabledCount": 0,
    "inputPortCount": 1,
    "outputPortCount": 1,
    "remoteSiteListeningPort": 10001,
    "siteToSiteSecure": false,
    "instanceId": "9d841c51-ab00-422e-811c-53c6dc2f8e59",
    "inputPorts": [
      {
        "id": "471deef6-2a6e-4a7d-912a-81cc17e3a204",
        "name": " From Node A",
        "comments": "",
        "state": "RUNNING"
      }
    ],
    "outputPorts": [
      {
        "id": "75f88005-0a87-4fef-8320-6219cdbcf18b",
        "name": "To A",
        "comments": "",
        "state": "STOPPED"
      }
    ]
  }
}